### ISSUE TITLE
Removing toastr by id not working ?

### ISSUE THREAD

**Issue by padsbanger** [diegoddox/react-redux-toastr#165]
``` js       
      dispatch(toastrActions.add({
          type: 'info',
          title: 'Action required.',
          id: 'someid',
          options: {
            showCloseButton: true,
            timeOut: 0,
            component: (
              <div>
                {intl.formatHTMLMessage({ id: 'youHaveNewBooking' })}
                <span
                  onClick={() => {
                    browserHistory.push('/dashboard/bookings');
                    toastrActions.remove('someid');
                  }}
                >See here</span>
              </div>
            ),
          },
        }));
```

Hello, I am building simple notifiaction system. When users sees the notification it can click on it, and it will close & redirect to the specific view. I am trying to close toastr by adding ID to it and removing with onClick, but it doesnt seem to work.

Any ideas ?

**Comment by diegoddox** [comment]
Hi @padsbanger sorry for my later reply, there is a bug while using the actions and not the emitter, but for the moment there is no solution for this.

**Comment by bmoquist** [comment]
@padsbanger - months later.... but there is a solution --- you forgot to dispatch the remove.

Here's how I've been handling it:
I only use the toastrActions.add() action, and unfortunately, the close button doesn't work well with this method  -- it only closes after the mouse leaves the toastr (which is baffling to the user since it doesn't actually close on the click -- and results in the user trying to click again and again). This is bug that should probably be fixed.

As a workaround, I add the onCloseButtonClick as I dispatch the toastr in my component. This will remove the toastr as soon as the user clicks. See below:

```js
export const addToastrCloseFunction = (toastrObj, dispatch) => {
    let toastrOptions = Object.assign({}, toastrObj.options, {
        onCloseButtonClick: () => {
          dispatch(toastrActions.remove(toastrObj.id))
        }
      }
    )

    return Object.assign({}, toastrObj, {
      options: toastrOptions
    }
  )
}

```

```js
const { myToastr } = toastrConstants
this.props.dispatch(addToastrCloseFunction(myToastr, this.props.dispatch))
```

**Comment by diegoddox** [comment]
A pr that may fix this issue #197 was merged.
